src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -264,9 +264,29 @@ def find_candidates(
             if ireq is not None:
                 ireqs.append(ireq)
 
+        for link in constraint.links:
+            # This part only has to convert an ireq from the requirements to a link.
+            # If there's only links, either it satisfies, or it doesn't.
```

Comment:
I don’t understand what this means, maybe it’s just me not being English-native.